oak-segment-tar/src/test/java/org/apache/jackrabbit/oak/segment/RecordCacheStatsTest.java

```diff
@@ -37,18 +37,9 @@ public class RecordCacheStatsTest {
     private final Random rnd = new Random();
     private final MemoryStore store = new MemoryStore();
 
-    private final RecordCache<Integer> cache = newRecordCache(KEYS);
+    private final RecordCache<Integer> cache = newRecordCache(KEYS * 4 / 3);
```

Comment:
I've noticed that if I remove your change and initialise the `RecordCache` with only KEYS (=100) values like before, then some tests fail. This means the behaviour of `RecordCache` is not preserved. This could break other code relying on the previous behaviour. 
Can you explain why the stats are not the same? Can you restore the previous behaviour?

Reply:
The reason for this change in behavior is that the Guava cache may preemptively evict items for performance reasons. This means that filling the cache exactly to capacity no longer results in an eviction count of 0. I cannot change this behavior. What I can do is keep both implementations like with the SegmentBufferWriterPool and only use the concurrent cache for compaction.

Reply:
Actually, there is another option. The Guava cache uses a `ConcurrentLinkedHashMap` internally. It would be possible to just use it manually. Downside of this approach is that I would also need to keep track of the `CacheStats` manually.

Reply:
I also think that keeping both implementations of the record cache is the safest option here.

Reply:
Nevermind the above. I restored the previous behavior by initializing the internal cache to 4/3 of the specified size. The old implementation also did this with its `LinkedHashMap`.